### PR TITLE
infrt add trt engine

### PR DESCRIPTION
**Pull request by jiweibo**
### PR types

Others
### PR changes

Others
### Describe

infrt add trt engine

### CODE REVIEW

File: paddle/infrt/backends/tensorrt/trt_engine.cc [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+    buffers[bind_index] = static_cast<void*>(bind.buffer->data<float>());
+  }
+
+  contexts_.front()->enqueueV2(buffers.data(), ctx.stream(), nullptr);
+}
+
+void TRTEngine::FreshDeviceId() {
+  int count;
+  cudaGetDeviceCount(&count);
+  assert(device_id_ < count);
```
Comment (shangzhizhou) [inline paddle/infrt/backends/tensorrt/trt_engine.cc]: CHECK(device_id_ < count) << "error message here".

Comment (jiweibo) [reply paddle/infrt/backends/tensorrt/trt_engine.cc]: done

File: paddle/infrt/backends/tensorrt/test_trt_engine.cc [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <math.h>
+
+#include "NvInfer.h"
```
Comment (Superjomn) [inline paddle/infrt/backends/tensorrt/test_trt_engine.cc]: The third party include please use  `<>` to distinguish from internal header files.

File: paddle/infrt/backends/tensorrt/trt_options.h [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "NvInfer.h"
```
Comment (Superjomn) [inline paddle/infrt/backends/tensorrt/trt_options.h]: include <NvInfer.h> 

全局

Comment (jiweibo) [reply paddle/infrt/backends/tensorrt/trt_options.h]: done

**Review by shangzhizhou — APPROVED** [review]
LGTM

**Pull request merged**